Priyankaben Patel/week_01/MTA/js/Homework.js

```diff
@@ -0,0 +1,109 @@
+The key to the lab is finding the index positions of each stop. (hint: indexOf())
+Make sure the stops that are the same for different lines have different names (i.e. 23rd on the N and on the 6 need to be differentiated)
+*/
+
+
+const subwayLines =[
+  {
+    id: 'N',
+    stops: ['Times Square', '34th', 'n28th', 'n23rd', 'Union Square', 'n8th'],
+    junction: 'Union Square'
+  },
+  {
+    id: 'L',
+    stops: ['l8th', '6th', 'Union Square', '3rd', '1st'],
+    junction: 'Union Square'
+  },
+  {
+    id:'six',
+    stops: ['Grand Central', '33rd', 'six28th', 'six23rd', 'Union Square', 'Astor Place'],
+    junction: 'Union Square'
+  }
+];
+
+
+  const planTrip = function (getInLine, getInStop, getOutLine, getOutStop) {
+    firstMetroLineStops = [];
+    secondMetroLineStops = [];
+    let idFound = false;
+    for (let i = 0; i < subwayLines.length; i++) {
+
+// check for getIn ID match in array of subwayLines
+    if (getInLine === subwayLines[i].id){
+      getInID = i;
+      firstTripLineId = subwayLines[i].id
+      firstTrip = subwayLines[i].stops;
+      junction = 'Union Square'
+      idFound = true;
+
+    }
+// Check for newLine ID and stops
+    if (getOutLine === subwayLines[i].id){
+      changeLineID = i;
+      secondTripLineId = subwayLines[i].id
+      secondTrip = subwayLines[i].stops;
+      idFound = true;
+      break;
+    }
+
+    if (!idFound) {
+      return false;
+    }
+  }
+// looping to get the firsttrip before reaching junction
+    for (let index = firstTrip.indexOf(getInStop) + 1; index <=firstTrip.indexOf('Union Square'); index ++) {
+      firstMetroLineStops.push(firstTrip[index]);
+
+    }
+    console.log('You must travel through the following stops on the ' + firstTripLineId + 'line: ' + firstMetroLineStops + '.')
+    console.log('Changed at ' + junction);
+
+// check for change to ID match in array of subwayLines and going backward
+    if (secondTrip.indexOf(getOutStop) < secondTrip.indexOf('Union Square')) {
+
+      for (let k = secondTrip.indexOf('Union Square') - 1; k >= secondTrip.indexOf(getOutStop); k --) {
+      secondMetroLineStops.push(secondTrip[k]);
+      }
+    }
+
+    else if (secondTrip.indexOf(getOutStop) > secondTrip.indexOf('Union Square')){
+// check for change to ID match in array of subwayLines and going forward
+      for (let a = secondTrip.indexOf('Union Square') + 1; a <= secondTrip.indexOf(getOutStop); a ++) {
+      secondMetroLineStops.push(secondTrip[a]);
+      }
+    };
+    console.log('Your journey continues through the following stops on ' + secondTripLineId + 'line: ' + secondMetroLineStops );
+    console.log(firstMetroLineStops.length + secondMetroLineStops.length + ' total stops');
+    return firstMetroLineStops;
+  }
+
+//planTrip('L', 'l8th', 'six', '33rd')
+planTrip('N', 'Times Square', 'L', '1st')
+planTrip('N', '34th', 'L', 'l8th')
+//planTrip('L', '34th', 'six', 'Astor Place')
```

Comment:
I find the following test cases not pass:
1. planTrip("six", "Astor Place", "N", "Times Square");
2. planTrip("six", "Union Square", "N", "Times Square");
3. planTrip("L", "1st", "six", "Union Square");